src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1018,7 +1018,7 @@ impl LinkCollector<'_, '_> {
             (link.trim(), None)
         };
 
-        if path_str.contains(|ch: char| !(ch.is_alphanumeric() || ":_<>, !".contains(ch))) {
+        if path_str.contains(|ch: char| !(ch.is_alphanumeric() || ":_<>, !*()[]&;".contains(ch))) {
```

Comment:
Won't this make us bail for tuples?

Reply:
What do you mean? This says "ignore the link unless it matches `a-z` or one of the whietlisted characters; this is explicitly adding `()` to the whitelist. Before, `()` would make it ignore the link.

Reply:
oh, i'm dumb.

Reply:
r=me once you think petrochenkov's review is satisfied!

Reply:
It's a sign this logic is too complicated and we need to simplify it :P